lib/matplotlib/axes/_base.py

```diff
@@ -483,6 +483,8 @@ def __init__(self, fig, rect,
             self._position = rect
         else:
             self._position = mtransforms.Bbox.from_bounds(*rect)
+        if self._position.width < 0 or self._position.height < 0:
+            raise ValueError('Height and width specified must be non-negative')
```

Comment:
Very minor point: probably "Width and height must be non-negative" is better (as the order is x, y).  I wouldn't add "specified" but I'll let a native speaker chime in :p